xbmc/Application.cpp

```diff
@@ -1359,26 +1358,15 @@ void CApplication::OnSettingChanged(std::shared_ptr<const CSetting> setting)
   {
     CheckOSScreenSaverInhibitionSetting();
   }
-  else if (StringUtils::StartsWithNoCase(settingId, "audiooutput."))
-  { // AE is master of audio settings and needs to be informed first
-    m_ServiceManager->GetActiveAE().OnSettingsChange(settingId);
-
-    if (settingId == CSettings::SETTING_AUDIOOUTPUT_GUISOUNDMODE)
-    {
-      m_ServiceManager->GetActiveAE().SetSoundMode((std::static_pointer_cast<const CSettingInt>(setting))->GetValue());
-    }
-    // this tells player whether to open an audio stream passthrough or PCM
```

Comment:
shouldn't this comment also be moved the few lines down to the new location of this condition? Might not harm to keep for others to understand why this callback is here and not in AE

Reply:
I don't comment the obvious. The comment did not provide more info than a skilled dev can derive from the method names.
btw: the comment was actually wrong :)

Reply:
ok. Btw, should the application have to deal with the passthrough setting at all from a design point of view? Or would it be cleaner for AE to fire this event, as it knows how to deal with this setting in first place? Just curious why this message is fired in cApp.

Reply:
This has ever been a hack. A clean design would be that AE askes its clients to reopen streams. In the meantime this hack fits better here than in AE.